mne_bids/path.py

```diff
@@ -1029,6 +1029,114 @@ def _check_non_sub_ses_entity(bids_path):
+
+    Returns
+    -------
+    str | None
+        If `return_str` is ``True``, the fields are returned as a
+        str. Else, ``None`` is returned and the fields are printed.
+    """
+    if not op.exists(folder):
+        raise ValueError('Directory does not exist: {}'.format(folder))
+
+    _validate_type(line_numbers, bool, 'line_numbers')
+    _validate_type(return_str, bool, 'return_str')
+    outfile = StringIO() if return_str else None
+
+    # Recursively walk through all directories
+    lim = 10
+    for root, _, files in os.walk(folder, topdown=True):
+        for file in files:
+            na_lines = list()
+            with open(op.join(root, file), 'r', encoding='utf-8-sig') as fid:
+                try:
+                    header = fid.readline()
+                except UnicodeDecodeError:
+                    continue
+                if '\t' in header:  # format tsv files nicely
+                    header = ''.join([str(val) + (lim - len(val)) * ' ' if
+                                      len(val) < lim else f'{val[:lim - 1]} '
+                                      for val in header.split('\t')])
+                if line_numbers:
+                    header = f'1    {header}'
+                header = header.rstrip()
+                for i, line in enumerate(fid):
+                    if 'n/a' in line:
+                        if '\t' in line:
+                            line = ''.join([
+                                str(val) + (lim - len(val)) * ' ' if
+                                len(val) < lim else f'{val[:lim - 1]} '
+                                for val in line.split('\t')])
+                        if line_numbers:
+                            line = f'{i + 2}    {line}'
+                        na_lines.append(line.rstrip())
+            if na_lines:
+                print(op.join(op.relpath(root, folder), file), file=outfile)
+                if op.splitext(file)[-1] == '.tsv':
+                    print(f'    {header}', file=outfile)
+                if len(na_lines) > 10:
+                    na_lines = na_lines[:10]
+                for line in na_lines:
+                    print(f'    {line}', file=outfile)
+
+    if outfile is not None:
+        return outfile.getvalue()
+
+
+def print_subject_sidecars(bids_path, return_str=False):
+    """Print the contents of all the files for one subject.
+
+    Parameters
+    ----------
+    bids_path : mne_bids.BIDSPath
+        The path with the ``bids_root`` and ``subject`` identifiers.
+    return_str : bool
+        If ``True``, return the text as a str instead of printing it.
+    """
+    subject_dir = op.join(bids_path.root, f'sub-{bids_path.subject}')
+    if not op.exists(subject_dir):
+        raise ValueError(f'Subject directory {subject_dir} does not exist')
+    _validate_type(return_str, bool, 'return_str')
+    outfile = StringIO() if return_str else None
+    for root, _, files in os.walk(subject_dir, topdown=True):
```

Comment:
using https://docs.python.org/3/library/pathlib.html#pathlib.Path.rglob seems much easier. I would these days encourage you as much as possible to use pathlib.Path